File: generated-code/example/scala/TcpReadWrite/src/main/scala/model/Structure.scala
package model

import util.StreamUtil

/**
 * Example structure
 *
 * @param text Text
 * @param floatNumber 32-bit float
 * @param doubleNumber 64-bit float
 */
case class Structure(text: String, floatNumber: Float, doubleNumber: Double) {
    /**
     * Write Structure to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeString(stream, text)
        StreamUtil.writeFloat(stream, floatNumber)
        StreamUtil.writeDouble(stream, doubleNumber)
    }

    /**
     * Get string representation of Structure
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Structure { ")
        stringBuilder.append("text: ")
        stringBuilder.append('"' + text + '"')
        stringBuilder.append(", ")
        stringBuilder.append("floatNumber: ")
        stringBuilder.append(floatNumber)
        stringBuilder.append(", ")
        stringBuilder.append("doubleNumber: ")
        stringBuilder.append(doubleNumber)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Structure {
    /**
     * Read Structure from input stream
     */
    def readFrom(stream: java.io.InputStream): Structure = Structure(
        StreamUtil.readString(stream),
        StreamUtil.readFloat(stream),
        StreamUtil.readDouble(stream)
    )
}

File: generated-code/java/example/TcpReadWrite/src/main/java/model/Structure.java
package model;

import util.StreamUtil;

/**
 * Example structure
 */
public class Structure {
    /**
     * Text
     */
    private String text;

    /**
     * Text
     */
    public String getText() {
        return text;
    }

    /**
     * Text
     */
    public void setText(String value) {
        this.text = value;
    }
    /**
     * 32-bit float
     */
    private float floatNumber;

    /**
     * 32-bit float
     */
    public float getFloatNumber() {
        return floatNumber;
    }

    /**
     * 32-bit float
     */
    public void setFloatNumber(float value) {
        this.floatNumber = value;
    }
    /**
     * 64-bit float
     */
    private double doubleNumber;

    /**
     * 64-bit float
     */
    public double getDoubleNumber() {
        return doubleNumber;
    }

    /**
     * 64-bit float
     */
    public void setDoubleNumber(double value) {
        this.doubleNumber = value;
    }

    public Structure(String text, float floatNumber, double doubleNumber) {
        this.text = text;
        this.floatNumber = floatNumber;
        this.doubleNumber = doubleNumber;
    }

    /**
     * Read Structure from input stream
     */
    public static Structure readFrom(java.io.InputStream stream) throws java.io.IOException {
        String text;
        text = StreamUtil.readString(stream);
        float floatNumber;
        floatNumber = StreamUtil.readFloat(stream);
        double doubleNumber;
        doubleNumber = StreamUtil.readDouble(stream);
        return new Structure(text, floatNumber, doubleNumber);
    }

    /**
     * Write Structure to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeString(stream, text);
        StreamUtil.writeFloat(stream, floatNumber);
        StreamUtil.writeDouble(stream, doubleNumber);
    }

    /**
     * Get string representation of Structure
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Structure { ");
        stringBuilder.append("text: ");
        stringBuilder.append('"' + text + '"');
        stringBuilder.append(", ");
        stringBuilder.append("floatNumber: ");
        stringBuilder.append(String.valueOf(floatNumber));
        stringBuilder.append(", ");
        stringBuilder.append("doubleNumber: ");
        stringBuilder.append(String.valueOf(doubleNumber));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}